### ISSUE TITLE
[Discuss] Support for bulk upload to API

### ISSUE THREAD

**Issue by majagrubic** [elastic/package-spec#406]
As you may know, the `shared-ux` team is working on sample data re-architecture, and we'd like to leverage EPR for that. (Previous discussion [here](https://github.com/elastic/package-spec/issues/346)). Right now, sample data is a (gzipped) json [file](https://github.com/elastic/kibana/blob/main/src/plugins/home/server/services/sample_data/data_sets/flights/flights.json.gz) with Elasticsearch documents that are then bulk-uploaded to ES when a user requests an install. 
If we were to add a new integration that would contain this sample data, what is the best way to make the bulk upload work?

cc @joshdover

**Comment by majagrubic** [comment]
As we discussed in the call today, we would also need to add support for a new package type that would be "asset only" or something similar. Let me know if you want me to open a new issue for that, or can we use this issue to coordinate.

**Comment by jsoriano** [comment]
I have added this issue to the implementation plan in https://github.com/elastic/package-spec/issues/346. I think we can continue there.

**Comment by jen-huang** [comment]
My initial thinking is that we can do something very similar to what sample data does today - perhaps we can even re-purpose the existing code.

That assumes that we are ok with storing archive files on the registry. @jsoriano the other discussion threads are a bit long - do you think that is ok?

This means that the package-spec will need to define enough information for Fleet to know that _filexyz.gzip_ contains ES documents and should be handled as such.

@kpollich, WDYT?

**Comment by jsoriano** [comment]
> That assumes that we are ok with storing archive files on the registry. @jsoriano the other discussion threads are a bit long - do you think that is ok?
> 
> This means that the package-spec will need to define enough information for Fleet to know that _filexyz.gzip_ contains ES documents and should be handled as such.

Yes, if we distribute these files through the registry, their format must be defined in the package-spec. This would be aligned with the implementation plan in the description of https://github.com/elastic/package-spec/issues/346, that summarizes the discussion threads.
